### PR TITLE
fix the generation of invalid/incorrect scripts for Next.js applications

### LINKED ISSUE #3654: 🐛 C3: `pages:build` script with pnpm scaffolding for Next does not work
### Which Cloudflare product(s) does this pertain to?

C3 v.2.0.14

### Describe the Bug

I've created a next application via 
```
pnpm create cloudflare
```

And the resulting package.json does use pnpm for its scripts:
Screenshot 2023-07-21 at 14 48 13

But the `pages:build` script does not work:
Screenshot 2023-07-21 at 14 51 10

The command should be
```
pnpm exec next-on-pages
```

Screenshot 2023-07-21 at 14 54 27

(same for `pages:watch`)
___

This is the relevant piece of code that needs updating:

https://github.com/cloudflare/workers-sdk/blob/99baf58b7c35e85c90e1f6df4cea841f31c0a709/packages/create-cloudflare/src/frameworks/next/index.ts#L90-L93

___

I'm opening this issue for pnpm but yarn (with its different versions) also needs to be tested to make sure that the script works properly with it

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
resolves #3654

### CODE REVIEW

File: packages/create-cloudflare/src/frameworks/next/index.ts [hunk 1]
```diff
@@ -140,11 +141,16 @@ const config: FrameworkConfig = {
...
 	displayName: "Next",
-	packageScripts: {
-		"pages:build": `${dlx} @cloudflare/next-on-pages@1`,
-		"pages:deploy": `${npm} run pages:build && wrangler pages deploy .vercel/output/static`,
-		"pages:watch": `${npx} @cloudflare/next-on-pages@1 --watch`,
-		"pages:dev": `${npx} wrangler pages dev .vercel/output/static --compatibility-flag=nodejs_compat`,
+	getPackageScripts: async () => {
+		const nextOnPagesScope = ["npm", "bun"].includes(npm) ? "@cloudflare/" : "";
+		const nextOnPagesCommand = `${nextOnPagesScope}next-on-pages`;
+		const pmCommand = npm === "npm" ? "npm run" : npm;
```
Comment (jculvey) [inline packages/create-cloudflare/src/frameworks/next/index.ts:147]: How about we add a `run` to `detectPackageManager` so we can reat it like `npx` and `dlx`?

File: packages/create-cloudflare/src/frameworks/next/index.ts [hunk 2]
```diff
@@ -140,11 +141,16 @@ const config: FrameworkConfig = {
 	generate,
 	configure,
 	displayName: "Next",
-	packageScripts: {
-		"pages:build": `${dlx} @cloudflare/next-on-pages@1`,
-		"pages:deploy": `${npm} run pages:build && wrangler pages deploy .vercel/output/static`,
-		"pages:watch": `${npx} @cloudflare/next-on-pages@1 --watch`,
```
[the remark this answers is not in the document]
Comment (dario-piotrowicz) [reply packages/create-cloudflare/src/frameworks/next/index.ts:146]: ah good spot! no this is wrong! my bad!

we need `npm run` for the `pages:deploy` script, not here 😓

File: packages/create-cloudflare/src/frameworks/next/index.ts (hunk 1 above)
Comment (dario-piotrowicz) [reply packages/create-cloudflare/src/frameworks/next/index.ts]: no sorry, the logic you reviewed was wrong, that's my bad sorry for the confusion 😓

**Comment by dario-piotrowicz** [comment]
@jculvey when reviewing this PR please have a look at the details I put here: https://github.com/cloudflare/workers-sdk/issues/3654#issuecomment-1766319602 as that shows what's the actual correct output

(PS: I am tempted to add unit tests to check that but it feels a bit of an overkill and inconsistent with our other frameworks so I am a bit hesitant in doing that 😕)

**Review by jculvey — APPROVED** [review]

**Comment by admah** [comment]
@dario-piotrowicz is this waiting on anything or can it be merged?

**Comment by dario-piotrowicz** [comment]
@admah yeah sorry about it 🙇, it can be merged, but I wanted to do one last double check to make extra sure that the changes are correct, I'll do that today and merge the PR if everything is as it should be, I hope that's fine

**Comment by admah** [comment]
That's great! I just wanted to make sure the PR was good and not forgotten 😄

**Comment by dario-piotrowicz** [comment]
I've tested it with npm and pnpm, bun doesn't seem able to run the package locally so I'll need to test it with the beta version after merging, I think we do have issues with yarn so that's also to be checked later

But I think this should be good 🙂 (even if there were to be something a bit off this is still a net improvement since the current scripts do not work anyways)

**Pull request merged**